funcx_sdk/funcx/serialize/concretes.py

```diff
@@ -124,16 +128,87 @@ class DillCode(SerializeBase):
     def __init__(self):
         super().__init__()
 
-    def serialize(self, data):
+    def serialize(self, data) -> str:
         x = codecs.encode(dill.dumps(data), "base64").decode()
         return self.identifier + x
 
-    def deserialize(self, payload):
+    def deserialize(self, payload: str):
         chomped = self.chomp(payload)
         function = dill.loads(codecs.decode(chomped.encode(), "base64"))
         return function
 
 
+class CombinedCode(SerializeBase):
+    """This method uses multiple methods to serialize a function
+
+    # These should be a superset of the methods
+    Code from interpreter/main        : Yes
+    Code from notebooks               : Yes
+    Works with mismatching py versions: Yes
+    Decorated fns                     : Yes
+    """
+
+    identifier = "10\n"
+
+    def get_multiple_payloads(self, payload: str) -> Tuple:
+        parts = self.chomp(payload).split(COMBINED_SEPARATOR)
+        assert len(parts) == 2 * len(DO_SERIALIZE_METHODS)
+        payloads = []
+        for index in range(len(parts) // 2):
+            payloads.append([parts[index], parts[index + 1]])
+
+        # ie. [['04\n', 'gACQ...'], ['01\n', 'sAAbk2...'], ...]
+        return payloads
+
+    def serialize(self, data) -> str:
+        chunks = []
+        for id in DO_SERIALIZE_METHODS:
+            method = METHODS_MAP_CODE[id]()
+            serialized = method.serialize(data)
+            id_length = len(id)
+            chunks.append(serialized[:id_length])
+            chunks.append(serialized[id_length:])
+        return self.identifier + COMBINED_SEPARATOR.join(chunks)
+
+    def deserialize(self, payload: str, variation: int = 1):
+        """
+        Return the other variants, if encoding contains them,
+        if requested.  varation is 1-indexed (ie. 2 means 2nd method)
+        """
+        count = 0
+        for serial_id, encoded_func in self.get_multiple_payloads(payload):
+            count += 1
+            if count == variation:
+                deserialize_method = METHODS_MAP_CODE[serial_id]()
```

Comment:
Is it possible that the payload being deserialized includes an invalid `serial_id`?  For example, if a bit flipped in transit?  If so, this would manifest as a weird `KeyError`.

Reply:
You are right, it would raise an non-obvious error.  I had thought that the existing code would also throw the same during deserialization if serial_id is unknown, but checking I see that it does raise an Invalid Header exception, so I will also change this.  (Also, this will help in case of backwards compatibility issues or changing of what methods are used in Combined)